qualtran/symbolics/types.py

```diff
@@ -46,6 +46,19 @@ def is_symbolic(self):
         return True
 
 
+@frozen
+class Length:
+    """Symbolic value for an object that has a length."""
+
+    bitsize: SymbolicInt
```

Comment:
this doesn't have to represent a bitwidth, right? it can be the length of anything. Perhaps `n`.